source/sema.h

```diff
@@ -1753,14 +2228,40 @@ class sema
 
     auto start(token const& t, int) -> void
     {
-        //  We currently only care to look at identifiers
-        if (t.type() != lexeme::Identifier) {
+        if (t.type() == lexeme::Dot) {
+            started_member_access = true;
+            started_this_member_access = *(&t - 1) == "this";
```

Comment:
I saw this trick in `emit(unqualified_id_node)`.
I suppose it exploits the contiguity of storage.

Reply:
Commit b37e32b5904454eb4ac88de81e37c11529627f00 is what it looked like before.
Should I revert that?

Reply:
OK, for now let's not let it be a distraction from the main thing which is to get this working. I see it's in a reviewable state so I'll plan to keep looking through the PR further before the weekend. Thanks!

Reply:
Applied your suggestion.
There remains one case, though:
`*(&t + 1) == "("`.